app/questionnaire/value_source_resolver.py

```diff
@@ -55,10 +55,13 @@ def _is_block_on_path(self, block_id: str) -> bool:
         if block_id in self.schema.list_collector_repeating_block_ids:
             # repeating blocks aren't on the path, so check the parent list collector
             list_name = self.schema.list_names_by_list_repeating_block_id[block_id]
-            # Type ignore: section and list collector will both exist if the block is repeating
+            # Type ignore: list collector will exist if the block is repeating
             section: ImmutableDict = self.schema.get_section_for_block_id(block_id)  # type: ignore
-            list_collector_block: ImmutableDict = self.schema.get_list_collector_for_list(section, list_name)  # type: ignore
-            return list_collector_block["id"] in self.routing_path_block_ids
+            list_collector_blocks: list = self.schema.get_list_collectors_for_list(for_list=list_name, section_id=section["id"])  # type: ignore
+
+            for list_collector_block in list_collector_blocks:
+                if list_collector_block["id"] in self.routing_path_block_ids:
+                    return True
```

Comment:
appreciate this is a horrible example and I cannot think of any reason why a schema would do this, but there might be other ways this edge case could happen

Reply:
Yeah wasn't sure about this, i think the previous logic was just checking the first list collector within the section

Reply:
Yeah I think it might be awkward to check in the current form, might be easier if we have something like `self._list_names_by_list_repeating_block_id` referring to the specific list collector

Reply:
Sounds like a good idea 👍

Reply:
Refactored now based on your suggestion 👍 although not even sure it needs a new method - just using the parent id map to lookup the list collector?

Reply:
yeah agreed, since its only once this looks like the way to go, nice